core/src/main/java/it/pagopa/selfcare/mscore/core/OnboardingDao.java

```diff
@@ -204,6 +249,24 @@ private void updateOrCreateUser(List<String> toUpdate, UserToOnboard userToOnboa
         }
     }
 
+    private Optional<OnboardedProduct> updateOrCreateUserWithState(List<String> toUpdate, UserToOnboard userToOnboard, Institution institution, OnboardingRequest request, String tokenId, RelationshipState state) {
+        try {
+            OnboardedUser onboardedUser = isNewUser(toUpdate, userToOnboard.getId());
```

Comment:
still naming: `isNewUser` sounds like a method that returns a boolean while it returns a full `OnboardedUser` object instead; maybe `getOnboardedUser` or `ensureOnboardedUser` fits better?

Reply:
ditto